Code/Framework/AzFramework/Platform/Common/UnixLike/AzFramework/Device/DeviceAttributesCommon_UnixLike.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+ */
+
+#include <sys/utsname.h>
+#include <AzFramework/Device/DeviceAttributes.h>
+
+namespace AzFramework
+{
+    DeviceAttributeDeviceModel::DeviceAttributeDeviceModel()
+    {
+        m_value = 0;
+        utsname systemInfo;
+        if (uname(&systemInfo) != -1)
+        {
+            m_value = systemInfo.machine;
+        }
+    }
+
+    DeviceAttributeRAM::DeviceAttributeRAM()
+    {
+        m_value = 0;
+
+        FILE* f;
+        azfopen(&f, "/proc/meminfo", "r");
+        if (f)
+        {
+            char buffer[256] = { 0 };
+            int total = 0;
+            while (fgets(buffer, sizeof(buffer), f))
+            {
+                if (azsscanf(buffer, "MemTotal: %d", &total))
+                {
+                    // meminfo displays memory in kB
+                    constexpr float div = 1024.f * 1024.f;
+                    m_value = aznumeric_cast<float>(total) / div;
+                }
+            }
+            fclose(f);
+        }

```

Comment:
This could use TextParser [ParseTextFile](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/TextParser.h#L62-L76) function instead?

Also if someone has a 4TiB of memory, then the int would overflow, as it would be `2^42 / 1024` == `2^40 / 2^10` == `2^32` which is 1 above the maximum value an unsigned int can represent. An AZ::u64 should be used instead

```suggestion
        constexpr AZ::IO::OpenMode readOpenMode = AZ::IO::OpenMode::ModeRead;
        if (AZ::IO::SystemFileStream systemFileStream("/proc/meminfo", readOpenMode); systemFileStream.IsOpen())
        {
            AZ::u64 memTotalKB{}
            auto parseTextEntry = [&memTotalKB](AZStd::string_view textEntry)
            {
                constexpr AZStd::string_view MemTotalKey = "MemTotal:";
                if (size_t offset = textExtry.find(MemTotalKey); offset != AZStd::string_view::npos)
                {
                    AZStd::string_view memTotalValueString = textEntry.substr(offset + MemTotalKey.size());
                    AZStd::from_chars_result result = AZStd::from_chars(memTotalValueString.begin()
                memTotalValueString.end(), memTotalKB);
                    // If the std::errc{} value is default, then the parsing occurred successfully and the mem total has been found
                    // so return false to halt iteration
                    return result.ec != AZStd::errc{}
                }
                // Line has been parsed successfully, with no mention of "MemTotal:" being found
                return true;
            };

            AZ::Settings::TextParserSettings parserSettings{ parseTextEntry };
            AZ::Settings::ParseTextFile(m_textStream, parserSettings);
            if (memTotalKB > 0)
            {
                // meminfo displays memory in KiB (Kibibytes, base 1024)
                // Convert to GiB
                constexpr double KiBtoGiB = 1024.f * 1024.f;
                m_value = aznumeric_cast<float>(static_cast<double>(memTotalKB) / KiBtoGiB);
            }
        }

```

Reply:
I went ahead with the suggestions for naming and fixing the overflows, but using ParseTextFile does seem a lot more complicated and I prefer the simpler version.